terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/runners.ts

```diff
@@ -70,27 +70,56 @@ export function resetRunnersCaches() {
   ssmParametersCache.reset();
 }
 
-export async function findAmiID(metrics: Metrics, region: string, filter: string, owners = 'amazon'): Promise<string> {
+// Helper function to get the current AWS account ID
+async function getCurrentAccountId(): Promise<string | null> {
+  try {
+    const sts = new STS();
+    const identity = await sts.getCallerIdentity().promise();
+    if (!identity.Account) {
+      console.warn('Unable to retrieve AWS account ID, falling back to Amazon-only images');
+      return null;
+    }
+    return identity.Account;
+  } catch (error) {
+    console.warn('Error retrieving AWS account ID:', error);
+    return null;
+  }
+}
+
+export async function findAmiID(
+  metrics: Metrics, 
+  region: string, 
+  filter: string, 
+  owners: string[] = ['amazon']
+): Promise<string> {
   const ec2 = new EC2({ region: region });
+  const accountId = await getCurrentAccountId();
+  const lfAccountId = '391835788720';
```

Comment:
why is lf account hardcoded? Seems like a bug, something that would possibly not work in all cases (ali deployed in LF, using an image from Meta)